File: utils/ai.ts
"use server"
import OpenAI from "openai";

const apiKey = process.env.OPENAI_API_KEY!
console.log('apikey: ', apiKey)

const client = new OpenAI({
    apiKey: apiKey,
});

const stripCitations = (s: string) =>
    s
        .replace(/\(\s*\[[^\]]*\]\([^)]*\)\s*\)/g, "")
        .replace(/\[([^\]]*)\]\([^)]*\)/g, "$1")
        .replace(/https?:\/\/\S+/g, "")
        .replace(/\(\s*\)/g, "")
        .replace(/[ \t]{2,}/g, " ")
        .replace(/\s+([.,;:])/g, "$1")
        .trim();

export async function webSearchSummary(
    pmt: string,
    data: object
): Promise<string> {
    const response = await client.responses.create({
        model: "gpt-4.1",
        tools: [{ type: "web_search" }],
        input: `Prompt: ${pmt}\nData: ${JSON.stringify(data, null, 2)}`,
    });

    return stripCitations(response.output_text ?? "");
}

export async function summary(
    pmt: string,
    data: object
): Promise<string> {
    const prompt = `Prompt: ${pmt}\nData: ${JSON.stringify(data, null, 2)}`;

    const response = await client.chat.completions.create({
        model: "gpt-4.1-mini",
        messages: [
            {
                role: "user",
                content: prompt,
            },
        ],
    });

    return response.choices[0].message.content ?? "";
}

File: ai.py
from openai import OpenAI
from dotenv import load_dotenv
import os

load_dotenv()

api_key = os.getenv("OPENAI_API_KEY")

print("API key found:", api_key is not None)



client = OpenAI(api_key=api_key)

def summary(pmt,data):
    prompt = f"Prompt: {pmt}Data: {data}"
    response = client.chat.completions.create(
        model="gpt-4.1-mini",
        messages=[
            {"role": "user", "content": prompt}
        ]
    )
    
    #return "Temporary"
    
    return response.choices[0].message.content

